src/hearings/containers/request-hearing/date-priority-hearing/date-priority-hearing.component.ts

```diff
@@ -100,12 +149,104 @@ export class DatePriorityHearingComponent implements OnInit {
+        const choosenLatestDate = moment(this.getDateFormatted(this.latestHearingFormGroup, this.latestHearingDate.id), HearingDateEnum.DefaultFormat);
+        const isPastEarliestDate = choosenEarliestDate.isBefore() || choosenEarliestDate.isSame(new Date(), 'd');
+        const isPastLatestDate = choosenLatestDate.isBefore() || choosenLatestDate.isSame(new Date(), 'd');
+        const isLatestBeforeEarliest = choosenEarliestDate > choosenLatestDate;
+        const isEarliestDateValid = choosenEarliestDate.isValid();
+        const isLatestHearingDate = choosenLatestDate.isValid();
+        if (!isInValidEarliestDate && isPastEarliestDate) {
+          this.validationErrors.push({ id: this.earliestHearingDate.id, message: HearingDatePriorityEnum.DatePastError });
+          this.earliestDateOfHearingError = { isInvalid: true, messages: [HearingDatePriorityEnum.DatePastError] };
+        } else if (!isInValidLatestDate && isPastLatestDate) {
+          this.validationErrors.push({ id: this.latestHearingDate.id, message: HearingDatePriorityEnum.DatePastError });
+          this.latestDateOfHearingError = { isInvalid: true, messages: [HearingDatePriorityEnum.DatePastError] };
+        } else if (isInValidEarliestDate || isInValidLatestDate || !isEarliestDateValid || !isLatestHearingDate) {
+          if ((this.earliestHearingFormGroup.dirty || (this.earliestHearingFormGroup.pristine && this.latestHearingFormGroup.pristine)) && (isInValidEarliestDate || !isEarliestDateValid)) {
+            this.validationErrors.push({ id: this.earliestHearingDate.id, message: HearingDatePriorityEnum.DateRangeError });
+            this.earliestDateOfHearingError = { isInvalid: true, messages: [HearingDatePriorityEnum.DateRangeError] };
+          }
+          if ((this.latestHearingFormGroup.dirty || (this.earliestHearingFormGroup.pristine && this.latestHearingFormGroup.pristine)) && (isInValidLatestDate || !isLatestHearingDate)) {
+            this.validationErrors.push({ id: this.latestHearingDate.id, message: HearingDatePriorityEnum.DateRangeError });
+            this.latestDateOfHearingError = { isInvalid: true, messages: [HearingDatePriorityEnum.DateRangeError] };
+          }
+        } else if (isEarliestDateValid && isLatestHearingDate && isLatestBeforeEarliest) {
+          this.validationErrors.push({ id: this.earliestHearingDate.id, message: HearingDatePriorityEnum.EarliestHearingDateError });
+          this.earliestDateOfHearingError = { isInvalid: true, messages: [HearingDatePriorityEnum.EarliestHearingDateError] };
+        }
+      }
+      // Hearing Priority check
+      if (!this.priorityForm.controls.priority.valid) {
+        this.hearingPriorityError = HearingDatePriorityEnum.PriorityError;
+        this.validationErrors.push({ id: this.priorities[0].key, message: HearingDatePriorityEnum.PriorityError });
+      }
+    }
```

Comment:
Method too long, could this be broken down to some sub validation check? Also it will be easier for unit test